app/src/main/java/com/example/poste/activities/FolderViewActivity.java

```diff
@@ -72,77 +50,85 @@ protected void onCreate(Bundle savedInstanceState) {
-//                            Toast.makeText(FolderViewActivity.this, "Share this folder", Toast.LENGTH_LONG).show();
-                        }
-                    },
-                    currentFolder.getPosts()
-            );
-            postRecyclerView.setAdapter(postAdapter);
-            registerForContextMenu(postRecyclerView);
-        } catch (APIException e) {
-            throw new RuntimeException(e);
-        }
-    }
+        editBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                Intent intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
+                startActivity(intent);
+            }
+        });
 
-    @Override
-    public boolean onContextItemSelected(MenuItem item) {
-        Post post = postAdapter.getLocalDataSetItem();
-        Intent intent = null;
-        switch (item.getItemId()) {
-            case R.id.ctx_menu_edit_post:
-                intent = new Intent(FolderViewActivity.this, EditPostActivity.class);
-                intent.putExtra("folderId", currentFolder.getId());
-                intent.putExtra("postId", post.getId());
+        deleteBut.setOnClickListener(view -> {
+            if (PosteApplication.getSelectedPost() != null) {
+                PosteApplication.getSelectedFolder().removePost(PosteApplication.getSelectedPost());
+                Intent intent = new Intent(FolderViewActivity.this, FolderViewActivity.class);
                 startActivity(intent);
-                break;
-            case R.id.ctx_menu_delete_post:
-                try {
-                    // Remove post from folder
-                    API.removePostFromFolder(post.getId(), currentFolder.getId());
+            }
+        });
 
-                    // Delete post
-                    API.deletePost(post.getId());
-                } catch (APIException e) {
-                    e.printStackTrace();
-                    Toast.makeText(FolderViewActivity.this, R.string.internal_error, Toast.LENGTH_LONG).show();
-                }
+        // Set folder name in title bar
+        folderName.setText(PosteApplication.getSelectedFolder().getName());
 
-                Intent newIntent = new Intent(FolderViewActivity.this, DashboardActivity.class);
-                startActivity(newIntent);
-                break;
+        // Remove empty text if posts exist in folder or remove buttons if the folder is empty
+        if (PosteApplication.getSelectedFolder().getPosts().size() > 0) {
+            emptyText.setVisibility(View.GONE);
+        }
+        else {
+            openBut.setVisibility(View.GONE);
+            editBut.setVisibility(View.GONE);
+            deleteBut.setVisibility(View.GONE);
         }
-        return true;
-    }
 
+        // Fill post view (Recycler View)
+        postRecyclerView.setLayoutManager(new LinearLayoutManager(FolderViewActivity.this));
```

Comment:
Nice usage of RecyclerView; was this easy to work with?

Reply:
Not to bad. still trying to figure out some of the cosmetic features.